packages/components/rating/src/rating-segment.tsx

```
@@ -0,0 +1,101 @@
+  let offset = Number(Math.floor(value) - 1 == index);
+
+  if (!isSingleSelection) {
+    offset = Math.floor(value) > index ? 1 : 0;
+    offset = Math.floor(value) == index ? value - Math.floor(value) : offset;
+  }
+
+  let offsetRTL = 1 - Number(Math.floor(value) - 1 == index);
+
+  if (!isSingleSelection) {
+    offsetRTL = Math.floor(value) > index ? 0 : 1;
+    offsetRTL = Math.floor(value) == index ? (offsetRTL = 1 - (value - Math.floor(value))) : offset;
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)